sisl/geom/surfaces.py

```diff
@@ -0,0 +1,116 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at https://mozilla.org/MPL/2.0/.
+import numpy as np
+
+from sisl._internal import set_module
+from sisl import Atom, Geometry, SuperCell
+
+__all__ = ['fcc_slab']
+
+
+@set_module("sisl.geom")
+def fcc_slab(alat, atoms, miller, size=None, vacuum=None, orthogonal=False, offset=0):
+    """ Construction of a surface slab from a face-centered cubic crystal
+
+    The slab layers are stacked along the z-axis
+
+    Parameters
+    ----------
+    alat : float
+        lattice constant of the fcc crystal
+    atoms : Atom
+        the atom that the crystal consists of
+    miller : str or array
+        Miller indices of the surface facet
+    size : 3-array, optional
+        slab size along the lattice vectors
+    vacuum : float, optional
+        distance added to the third lattice vector to separate
+        the slab from its periodic images
+    orthogonal : bool, optional
+        if True returns an orthogonal lattice
+    offset : int, optional
+        index to shuffle the layer sequence, eg ABC to BCA
+    """
+    if isinstance(miller, int):
+        miller = str(miller)
+
+    if isinstance(miller, str):
+        miller = (int(miller[0]), int(miller[1]), int(miller[2]))
```

Comment:
we should check that `len(miller) == 3`

Reply:
Is this necessary? If `miller` does not match the implemented cases an error is raised anyways.

Reply:
Yes, but passing `miller='1111'` should fail I think.

Reply:
> Yes, but passing `miller='1111'` should fail I think.

and `miller='11'` should also fail. That would raise an `IndexError` which is somewhat counter intuitive, I think.

Reply:
Done!

Reply:
Done!